src/ProductConstructionService/Microsoft.DotNet.ProductConstructionService.Client/Build.cs

```diff
@@ -40,16 +41,47 @@ public string GetCommitLink()
             {
                 throw new InvalidOperationException($"Cannot get commit link of build with id {Id} because it does not have a repo URL.");
             }
-            string baseUrl = repoUrl.TrimEnd('/');
-            if (baseUrl.Contains("github.com"))
+            
+            if (!Uri.TryCreate(repoUrl, UriKind.Absolute, out Uri parsedUri))
             {
-                return $"{baseUrl}/commit/{Commit}";
+                throw new InvalidOperationException($"Failed to construct a commit link for build with id {Id} with repo url {repoUrl}.");
             }
-            else if (baseUrl.Contains("dev.azure.com"))
+
+            return parsedUri.Host switch
```

Comment:
You are still duplicating logic that is already in `GitRepoUrlUtils.cs`. You could use `GetRepoAtBranchUri` or `GetCommitLinkUri` directly in `Build.cs`.